dashboard/dashboard-ops-reverse-proxy.md

```diff
@@ -163,38 +219,89 @@ server_configs:
 >
 > After the modified and customized path prefix takes effect, you cannot directly access TiDB Dashboard. You can only access TiDB Dashboard through a reverse proxy that matches the path prefix.
 
-### Step 2: Modify the NGINX reverse proxy configuration
+### Step 2: Modify the reverse proxy configuration
+
+<details>
+<summary> <strong>Use HAProxy</strong> </summary>
+
+Taking `http://example.com:8033/foo/` as an example, the corresponding HAProxy configuration is as follows:
+
+{{< copyable "" >}}
+
+```haproxy
+frontend tidb_dashboard_front
+  bind *:8033
+  use_backend tidb_dashboard_back if { path /foo } or { path_beg /foo/ }
+
+backend tidb_dashboard_back
+  mode http
+  http-request set-path %[path,regsub(^/foo/?,/dashboard/)]
+  server tidb_dashboard 192.168.0.123:2379
+```
+
+Replace `192.168.0.123:2379` with IP and port of the actual address of the TiDB Dashboard obtained in [Step 1](#step-1-get-the-actual-tidb-dashboard-address).
+
+> **Warning:**
+>
+> You must retain the `if` part in the `use_backend` directive to ensure that services **only in this path** are behind reverse proxy; otherwise, security risks might be introduced. See [Secure TiDB Dashboard](/dashboard/dashboard-ops-security.md).
+
+If you want that the TiDB Dashboard service is run in the root path (such as `http://example.com:8033/`), use the following configuration:
+
+```haproxy
+frontend tidb_dashboard_front
+  bind *:8033
+  use_backend tidb_dashboard_back
+backend tidb_dashboard_back
+  mode http
+  http-request set-path /dashboard%[path]
+  server tidb_dashboard 192.168.0.123:2379
+```
+
+Modify the configuration and restart HAProxy for the modified configuration to take effect.
 
-Taking `http://example.com/foo` as an example, the corresponding NGINX configuration is as follows:
+</details>
+
+<details>
+<summary> <strong>Use NGINX</strong> </summary>
+
+Taking `http://example.com:8033/foo/` as an example, the corresponding NGINX configuration is as follows:
 
 {{< copyable "" >}}
 
 ```nginx
 server {
-  listen 80;
+  listen 8033;
   location /foo/ {
     proxy_pass http://192.168.0.123:2379/dashboard/;
   }
 }
 ```
 
+Replace `http://192.168.0.123:2379/dashboard/` with the actual address of the TiDB Dashboard obtained in [Step 1](#step-1-get-the-actual-tidb-dashboard-address).
+
 > **Warning:**
 >
-> Keep the `/dashboard/` path in the `proxy_pass` directive to ensure that only the services under this path are reverse proxied. Otherwise, security risks will be introduced. See [Secure TiDB Dashboard](/dashboard/dashboard-ops-security.md).
+> You must retain the `if` part in the `use_backend` directive to ensure that services **only in this path** are behind reverse proxy; otherwise, security risks might be introduced. See [Secure TiDB Dashboard](/dashboard/dashboard-ops-security.md).
```

Comment:
```suggestion
> You must retain the `/dashboard/` path in the `proxy_pass` directive to ensure that services **only in this path** are behind reverse proxy; otherwise, security risks might be introduced. See [Secure TiDB Dashboard](/dashboard/dashboard-ops-security.md).
```
from https://github.com/pingcap/docs-cn/pull/3828/files#diff-753d59b99895e7d37c715aa6c59ecb8bR250